src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,42 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int AGE_MIN = 18;
+    private static final int AGE_MAX = 122;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationServiceException {
+        if (user.getLogin() == null) {
+            throw new RegistrationServiceException("Error, login is " + user.getLogin());
+        }
+        if (user.getLogin().equals("")) {
+            throw new RegistrationServiceException("Error, login is " + user.getLogin());
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationServiceException("Error, password is " + user.getPassword());
+        }
+        if (user.getPassword().equals("")) {
+            throw new RegistrationServiceException("Error, password is " + user.getPassword());
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationServiceException("Error, age is " + user.getAge());
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationServiceException("User with this login is already exists");
+        }
+        if (user.getAge() < AGE_MIN) {
+            throw new RegistrationServiceException("User's age "
+                    + user.getAge() + " is less than 18 years");
+        }
+        if (user.getAge() > AGE_MAX) {
+            throw new RegistrationServiceException("User's age "
+                    + user.getAge() + " is more than 122 years");
```

Comment:
we have age limits as class constants, use them here please so if we change any of it in 1 place we wouldn't change  all through the code